finagle-core/src/test/scala/com/twitter/finagle/netty3/Netty3TransporterTest.scala

```diff
@@ -189,7 +189,7 @@ class Netty3TransporterTest extends FunSpec with MockitoSugar {
     }
 
     describe("SslHandler") {
-      it ("should close the channel if the remote peer closed TLS session") {
+      def mockSslEngine = {
```

Comment:
I looked at the failed assert:

```java
  assert !inNetBuf.hasRemaining() || engine.isInboundDone();
```

I'm not sure we run the tests with assertion enabled. So, could you please add the following configuration to the mock engine:

```java
when(engine.isInboundDone) thenReturn true
```

and check the `sbt +test` locally?